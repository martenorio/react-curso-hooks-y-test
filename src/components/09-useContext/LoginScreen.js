import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {
  // obtener referencia al context
  const {user, setUser} = useContext(UserContext)
  //data para setUser
  const loginUser = {
    id:423234,
    name:'Miguel Angel Ramirez Tenorio',
  }
  return (
    <div>
        <h1>LoginScreen</h1>
        <hr />
        <button 
        className='btn btn-primary' 
        onClick={() => setUser(loginUser)}
        >
          login
        </button>
    </div>
  )
}
