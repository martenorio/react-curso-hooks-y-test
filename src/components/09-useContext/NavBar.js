import React from 'react'
import { Link, NavLink } from "react-router-dom";
export const NavBar = () => {
    return (
        // <nav>
        //     <ul>
        //         <li><Link to="./">home</Link> </li>
        //         <li><Link to="./about">about</Link></li>
        //         <li><Link to="./login">login</Link></li>
        //     </ul>
        // </nav>
        <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
            <div className="container-fluid">
                <Link className="navbar-brand" to="./">React Router</Link> 
                <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>
                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        <NavLink to="./" exact="true" className={({ isActive }) => (!isActive ? "nav-link" : "nav-link active")}>home</NavLink > 
                        <NavLink to="./about" exact="true" className={({ isActive }) => (!isActive ? "nav-link" : "nav-link active")}>about</NavLink >
                        <NavLink to="./login" exact="true" className={({ isActive }) => (!isActive ? "nav-link" : "nav-link active")}>login</NavLink >
                    </div>
                </div>
            </div>
        </nav>
    )
}
