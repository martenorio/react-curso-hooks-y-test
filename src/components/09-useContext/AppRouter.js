import React from 'react'
import {
    BrowserRouter as Router,
    Routes,
    Route,
    Navigate
  } from "react-router-dom";
  import { NavBar } from './NavBar';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
  return (
    <Router>
        <div>
            <NavBar />
            <div className="container">
              <Routes>
                  <Route exact path='/' element={ <HomeScreen /> }></Route>
                  <Route exact path='/about' element={ <AboutScreen />}></Route>
                  <Route exact path='/login' element={ <LoginScreen />}></Route>
                  <Route path="*" element={ <Navigate replace to="/" /> /*<HomeScreen /> */ }></Route>
              </Routes>
            </div>
        </div>
    </Router>
  )
}
