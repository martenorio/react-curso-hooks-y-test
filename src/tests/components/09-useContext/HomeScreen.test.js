import { mount, shallow } from "enzyme";
import React from "react";
import { HomeScreen } from "../../../components/09-useContext/HomeScreen";
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('HomeScreen tests', () => {
    const user = {
        name:"miguel",
        email:"m@gmail.com"
    }
    const wrapper = mount(
        <UserContext.Provider value={ {user} }>
            <HomeScreen />
        </UserContext.Provider>
    )
    test('should display correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
})