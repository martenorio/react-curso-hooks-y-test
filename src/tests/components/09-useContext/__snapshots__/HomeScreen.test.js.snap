// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen tests should display correctly 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "miguel",
   "email": "m@gmail.com"
}
  </pre>
</div>
`;
